### DISCUSSION TITLE
Error 403: rate_limit_exceeded

### DISCUSSION THREAD

**Discussion by Adam-Lewis100** [kewisch/gdata-provider#376] [Support] [answered]
Hi,
I've been using the Provider for years without issue but have now received this pop-up message: "Authorisation Error Error 403: rate_limit_exceeded  This app has reached its sign-in rate limit for now.". 
Cheers,
Adam

**Comment by ItsReallyMeDE** [comment]
Same here. All google calendar connections to my TB instances using this tool have been cut (new TB version??) 

Assuming that this happened to many users, maybe the massive amount of necessary new authorisations has triggered the error?

**Reply by srsbiz** [reply]
Looks that way, just restarted TB and this time Google login worked on first try

**Reply by Adam-Lewis100** [reply]
Just tried again and the problem is resolved. I think ItsReallyMeDE is probably right - all users were forced to re-authenticate and it triggered the 403. Or maybe it's just Monday morning . . .

**Reply by silversheart** [reply]
Unfortunately it's not working for me :( After restarting TB, error 403: rate_limit_exceeded did not disapрear :,-(

**Reply by houstonwb** [reply]
Worked for me last night.  Authorization did not persist today.  One account worked after reauthorizing this morning.  Other account authorization failed twice with an hour in between attempts.  
Existing calendars did not update without Provider despite suggestions that Provider is no longer needed by Thunderbird.

**Reply by bgrinstein** [reply]
Restart did not work for me

**Reply by GWdd** [reply]
So far, restarting TB has had no effect - deleting calendars and reinstalling has had no effect; basically dead in the water. 

I like Thunderbird, but this Google gmail and Google calendar login trouble is becoming a usability issue. I know it is Google, but it still affects TB users.

**Reply by pfeiffep** [reply]
Still happening on Win10 + Ubuntu Mate

**Reply by pfeiffep** [reply]
Release notes has this info
"Re-use same key as 91.0.2 to avoid rate limit exceeded errors"

**Reply by jvaccaro** [reply]
As of 8pm, EST the provider-for-google-calendar plugin is still failing to authenticate with Google complaining that **_Error 403: rate_limit_exceeded_**
Seems Philipp Kewisch aka @kewisch in Hamburg is aware of the issue. See attached screenshot from the provider-for-google-calendar at googlegroups.com page: [https://groups.google.com/g/provider-for-google-calendar]
)

JohnV
Don't be nervous. Work calmly, joyously, recklessly on whatever is at hand. - Henry Mille
![Screenshot from 2022-01-24 20-04-25](https://user-images.githubusercontent.com/153742/150891371-4db04e6d-f893-4353-86ae-fce09bf12e8b.png)
r

**Comment by ItsReallyMeDE** [comment]
Restart does not work for me either.  It is not really a surprise, as the "error" is a "denial" on the side of Google and has not much to do with TB - I think....    Only time might help - or someone knowing about the details of this denial and how to avoid it ;-)

**Reply by Adam-Lewis100** [reply]
It may be a case of getting lucky and trying it when there is a lull in others trying to fix it. I'd think the limit applies to ones local Google server and so no need to wait for the USA to go to bed unless you live there.

**Comment by st333v** [comment, marked as answer]
in case anyones wondering, theres an open issue for this bug here: https://github.com/kewisch/gdata-provider/issues/373 which the developer has responded to

**Comment by atscomms** [comment]
This has been happening to me as well. It happened just now, for all of my calendar accounts

**Comment by lphilpot** [comment]
Still happening for me with TB 91.5.0 on Xubuntu. Maybe I need to just use my calendar on the web instead of in TB...

**Reply by Karen-KISSWPW** [reply]
Yes, I've started keeping a tab open in my browser with my gmail calendar open and linked all my calendars together. It was a bit of a pain to do but it was better than missing any meetings. I haven't been able to connect since yesterday.

**Reply by lphilpot** [reply]
Based on the info above, I'll just disable the extension until it's fixed.

**Comment by bob-atkins** [comment]
Same problem here.

Version 91.0.4
Last Updated January 24, 2022

Everything was working fine for the last few months until your most recent update.

I've tried everything. Restart TB, Uninstall the add-on (and delete data), restart, re-install add-on.

Remove app permissions in my Google account and then try to re-authenticate. Nada.

I get the same problem that I see many are reporting.

Authorization Error
Error 403: rate_limit_exceeded

This app has reached its sign-in rate limit for now.

Google limits how quickly an app can get new users. You can try signing in again later, or ask the developer (provider-for-google-calendar@googlegroups.com) to increase this app's sign-in rate limit.

If you are the developer for this app, you can request a sign-in rate limit increase

**Comment by RobinThai** [comment]
> 

Same problem here. I would like to provide detail that I've got from Google's pop-up message:

=============================
Google pop-up message:
Authorization Error
Error 403: rate_limit_exceeded

This app has reached its sign-in rate limit for now.

Google limits how quickly an app can get new users. You can try signing in again later, or ask the developer (provider-for-google-calendar@googlegroups.com) to increase this app's sign-in rate limit.

If you are the developer for this app, you can request a sign-in rate limit increase.
Learn more
The content in this section has been provided by the app developer. This content has not been reviewed or verified by Google.
If you’re the app developer, make sure that these request details comply with Google policies.

    login_hint: xx@gmail.com
    hl: en-US
    response_type: code
    redirect_uri: urn:ietf:wg:oauth:2.0:oob:auto
    client_id: 647770275422-383s2no7f243rp9iv4m31llptt09tld2.apps.googleusercontent.com
    access_type: offline
    scope: https://www.googleapis.com/auth/calendar https://www.googleapis.com/auth/tasks
============

**Comment by moanan** [comment]
Same here. MacOS 12.0.1.

**Comment by digitalit** [comment]
I have no problem in Thunderbird on MacOS 12.1 but i have only 3 calendars.

My customer with this problem has like 6 or 7 calendars, might be related to the amount of calendars?

**Comment by jvaccaro** [comment]
As of 7:25am, EST the provider-for-google-calendar plugin, version 91.0.4 has successfully authenticated with Google for all seven of my Google Calendars.

Thanks Philipp for an awesome plugin and the **above and beyond** dealing with this issue!

JohnV
_No good deed goes unpunished_

**Comment by kmg454** [comment]
The same happens with the Protopage adons for the google Calendar on Firefox while the adon works fine on Chrome

**Comment by widdi1977** [comment]
same here. win10prof
TB 91.5.1 (64-Bit)
Version 91.0.4

I use it for the calendar of my wife. I get the same error - but the calendar will be shown. If it shows her updated/new events, I have not tested yet

**Comment by lphilpot** [comment]
Still failing here as of 9:21 AM CST.

*Len Philpot*
***@***.***
/Sent from Thunderbird on Xubuntu Linux/

**Comment by GWdd** [comment]
Besides removing Google Provider - the applet crashing the process - is there a simple way to tell Thunderbird to not try to load calendars?

Five attempts today all - so far - using 2-part authentication transferring numbers between devices, multiple time each TB start.. it's getting old.

And yeah - tried the restart, reload, remove and replace, and remove and replace with older version.  Still receiving the 403 error.

**Comment by lphilpot** [comment]
I just disabled the extension in TB: Tools | Add-ons and Themes | 
Extensions and toggle the switch to the right on the Provider line. I 
canceled when it asked to unsubscribe and it just turned the extension 
off. Turn it back on the same way.

*Len Philpot*
***@***.***
/Sent from Thunderbird on Xubuntu Linux/

**Comment by glsf91** [comment]
Also have still the Error 403: rate_limit_exceeded

**Comment by GWdd** [comment]
Any chance this has anything to do with 2-key security being enabled - the thing that breaks POP email to Google servers?  I know I set that back in November, but the calendar has been working. Though POP email to Gmail servers broke hard - IMAP can still work using OAUTH.

**Comment by marcioschmitt** [comment]
Same problema here.
Two computers
Win10 & Win7
TB last version
